innersource-license.md

```diff
@@ -0,0 +1,83 @@
+## Title
+
+InnerSource License
+
+## Patlet
+
+Two companies that belong to the same group want to share software source code with each other but they are concerned about the implications in terms of legal liabilities or cross-company accounting. An **InnerSource License** provides a reusable legal framework for the inter-company agreement within the group, which opens up new collaboration options.
+
+## Problem
+
+When two companies within a group want to share code with each other, they need an agreement about the terms and often a legal contract. Creating such agreements on a per project basis takes effort and creates a barrier for sharing. i.e. a team/company might decide not to share their source code with another company in the group because it seems complicated.
+
+Barriers for sharing can lead to silos and duplication of effort in rebuilding similar solutions in multiple parts of the group.
+
+At the time of sharing the source code, it can not be reliably predicted what the value of sharing will be. Hence if the activity of sharing has any significant cost, the companies are unlikely to do it, as they are concerned about the return on investment.
+
+## Context
+
+- A large group with many companies (subsidiaries) that want to share code. (when the group gets larger, the value of this pattern increases)
+- These companies are legally independent units.
+- Multiple of these companies/subsidiaries are developing software, and are using services of the other companies. They have a motivation to contribute to each other’s source code.
+- Certain size of developers (crowdsourcing and ecosystem effects of InnerSource)
+
+## Forces
+
+- **Level of effort** required to write formal agreements, especially if they need to take into account technical, legal, and business perspectives.
+- Enterprises have many **internal regulations,** any agreements have to comply with these regulations, e.g. security, privacy, procurement processes, etc. The volume of regulations can make it difficult to assess whether sharing software between two companies is compliant with these regulations, especially when there is no standard procedure.
+- If any of the companies in the group has a **business model** that depends on proprietary code and accounting of licensing fees within the group
+- **Company culture** that isn’t used to sharing code
```

Comment:
My first reaction was: "Well, that's a general InnerSource force, so maybe we should drop it." But the more I think about it, I think it was right to list it here: Missing routine with sharing / collaborating, might increase the need for a license.

Maybe we should add a sentence though, to explain how this force ties into the pattern?

Reply:
Do you mean the whole section you marked or only **Company culture**?

Reply:
(Intended to only mark `Company culture`)

To clarify the suggestion: Make explicit how `company culture` plays into this pattern:
```suggestion
- **Company culture** that isn’t used to InnerSource collaboration and sharing code. This results in uncertainty about the rights and obligations when using shared code.
```

One could very well argue: "Whoops. Doesn't that drift too much into the problem section?" but I think it is important to have a tie in between the forces and other sections where possible.

Reply:
I will merge your suggestion in. This also points to the "psychological uncertainty" of the individual developer a little bit. So at least we have a note of it in here that we can elaborate on later if we want to.